maintainers/scripts/update.py

```diff
@@ -12,80 +14,150 @@
 
 class CalledProcessError(Exception):
     process: asyncio.subprocess.Process
+    stderr: Optional[bytes]
 
 class UpdateFailedException(Exception):
     pass
 
-def eprint(*args, **kwargs):
-    print(*args, file=sys.stderr, **kwargs)
-
-async def check_subprocess(*args, **kwargs):
+T = TypeVar('T')
+
+def make_object_from_json_value(value: Any, t: type[T]) -> T:
+    return cast(T, jsons.load(value, t))
```

Comment:
If you know a nicer data validation library, please let me know.

- `jsons` has [nice docs](https://jsons.readthedocs.io/en/latest/index.html) but does not have type library stubs or py.typed marker
- `json-strong-typing` is not packaged
- [`dataclasses-json`](https://github.com/lidatong/dataclasses-json) has ugly API and types are [currently broken](https://github.com/lidatong/dataclasses-json/pull/371)
- `pydantic` has [baroque API](https://docs.pydantic.dev/usage/dataclasses/) and does [automatic coercion](https://docs.pydantic.dev/usage/types/#strict-types) :nauseated_face:

Reply:
[`cattrs`](https://catt.rs/)?